### ISSUE TITLE
Fail to install spago@next on macOS Sonoma 14.5

### ISSUE THREAD

**Issue by FranklinChen** [purescript/spago#1229]
```
$ npm i -g spago@next
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm error code 1
npm error path /opt/homebrew/lib/node_modules/spago/node_modules/better-sqlite3
npm error command failed
npm error command sh -c prebuild-install || node-gyp rebuild --release
npm error TOUCH ba23eeee118cd63e16015df367567cb043fed872.intermediate
npm error   ACTION deps_sqlite3_gyp_locate_sqlite3_target_copy_builtin_sqlite3 ba23eeee118cd63e16015df367567cb043fed872.intermediate
npm error   TOUCH Release/obj.target/deps/locate_sqlite3.stamp
npm error   CC(target) Release/obj.target/sqlite3/gen/sqlite3/sqlite3.o
npm error   LIBTOOL-STATIC Release/sqlite3.a
npm error   CXX(target) Release/obj.target/better_sqlite3/src/better_sqlite3.o
npm error rm ba23eeee118cd63e16015df367567cb043fed872.intermediate
npm error prebuild-install warn install No prebuilt binaries found (target=22.3.0 runtime=node arch=arm64 libc= platform=darwin)
npm error gyp info it worked if it ends with ok
npm error gyp info using node-gyp@10.1.0
npm error gyp info using node@22.3.0 | darwin | arm64
npm error gyp info find Python using Python version 3.12.3 found at "/opt/homebrew/opt/python@3.12/bin/python3.12"
npm error gyp info spawn /opt/homebrew/opt/python@3.12/bin/python3.12
npm error gyp info spawn args [
npm error gyp info spawn args '/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm error gyp info spawn args 'binding.gyp',
npm error gyp info spawn args '-f',
npm error gyp info spawn args 'make',
npm error gyp info spawn args '-I',
npm error gyp info spawn args '/opt/homebrew/lib/node_modules/spago/node_modules/better-sqlite3/build/config.gypi',
npm error gyp info spawn args '-I',
npm error gyp info spawn args '/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm error gyp info spawn args '-I',
npm error gyp info spawn args '/Users/chen/Library/Caches/node-gyp/22.3.0/include/node/common.gypi',
npm error gyp info spawn args '-Dlibrary=shared_library',
npm error gyp info spawn args '-Dvisibility=default',
npm error gyp info spawn args '-Dnode_root_dir=/Users/chen/Library/Caches/node-gyp/22.3.0',
npm error gyp info spawn args '-Dnode_gyp_dir=/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp',
npm error gyp info spawn args '-Dnode_lib_file=/Users/chen/Library/Caches/node-gyp/22.3.0/<(target_arch)/node.lib',
npm error gyp info spawn args '-Dmodule_root_dir=/opt/homebrew/lib/node_modules/spago/node_modules/better-sqlite3',
npm error gyp info spawn args '-Dnode_engine=v8',
npm error gyp info spawn args '--depth=.',
npm error gyp info spawn args '--no-parallel',
npm error gyp info spawn args '--generator-output',
npm error gyp info spawn args 'build',
npm error gyp info spawn args '-Goutput_dir=.'
npm error gyp info spawn args ]
npm error gyp info spawn make
npm error gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm error ./src/util/macros.lzz:150:35: error: no matching member function for call to 'SetAccessor'
npm error         recv->InstanceTemplate()->SetAccessor(
npm error         ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
npm error /Users/chen/Library/Caches/node-gyp/22.3.0/include/node/v8-template.h:1049:8: note: candidate function not viable: no known conversion from 'v8::AccessControl' to 'PropertyAttribute' for 5th argument
npm error   void SetAccessor(
npm error        ^
npm error /Users/chen/Library/Caches/node-gyp/22.3.0/include/node/v8-template.h:1055:8: note: candidate function not viable: no known conversion from 'v8::AccessorGetterCallback' (aka 'void (*)(Local<String>, const PropertyCallbackInfo<Value> &)') to 'AccessorNameGetterCallback' (aka 'void (*)(Local<Name>, const PropertyCallbackInfo<Value> &)') for 2nd argument
npm error   void SetAccessor(
npm error        ^
npm error ./src/objects/database.lzz:180:21: warning: variable 'status' set but not used [-Wunused-but-set-variable]
npm error                 int status = sqlite3_db_config(db_handle, SQLITE_DBCONFIG_ENABLE_LOAD_EXTENSION, 1, NULL);
npm error                     ^
npm error 1 warning and 1 error generated.
npm error make: *** [Release/obj.target/better_sqlite3/src/better_sqlite3.o] Error 1
npm error gyp ERR! build error 
npm error gyp ERR! stack Error: `make` failed with exit code: 2
npm error gyp ERR! stack at ChildProcess.<anonymous> (/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
npm error gyp ERR! System Darwin 23.5.0
npm error gyp ERR! command "/opt/homebrew/Cellar/node/22.3.0/bin/node" "/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--release"
npm error gyp ERR! cwd /opt/homebrew/lib/node_modules/spago/node_modules/better-sqlite3
npm error gyp ERR! node -v v22.3.0
npm error gyp ERR! node-gyp -v v10.1.0
npm error gyp ERR! not ok
npm error A complete log of this run can be found in: /Users/chen/.npm/_logs/2024-06-13T18_04_04_189Z-debug-0.log

```

**Comment by FranklinChen** [comment]
Apparently using node@20 does work. Perhaps this should be documented.

**Comment by f-f** [comment]
It's a pretty new thing, so I am still trying to figure out if it's a Node bug and/or if we can do anything about it. Any help would be appreciated!

**Comment by FranklinChen** [comment]
I noticed that some dependencies are very old, and that might well be the reason newer versions of Node can't build spago. You might want to update some of the dependencies:

```
$ ncu
Checking /Users/chen/spago/package.json
[====================] 18/18 100%

 better-sqlite3          ^8.6.0  →  ^11.0.0
 fast-glob              ^3.2.11  →   ^3.3.2
 fs-extra               ^10.0.0  →  ^11.2.0
 fuse.js                 ^6.5.3  →   ^7.0.0
 glob                    ^7.1.6  →  ^10.4.1
 markdown-it            ^12.0.4  →  ^14.1.0
 micromatch              ^4.0.5  →   ^4.0.7
 open                    ^9.1.0  →  ^10.1.0
 punycode                ^2.3.0  →   ^2.3.1
 semver                  ^7.3.5  →   ^7.6.2
 spdx-expression-parse   ^3.0.1  →   ^4.0.0
 ssh2                   ^1.14.0  →  ^1.15.0
 supports-color          ^9.2.3  →   ^9.4.0
 tar                    ^6.1.11  →   ^7.2.0
 tmp                     ^0.2.1  →   ^0.2.3
 yaml                    ^2.1.1  →   ^2.4.5

Run ncu -u to upgrade package.json
```

**Comment by FranklinChen** [comment]
In particular, just bumping up the most out-of-date dependency, `better-sqlite3`, enabled `npm` on Node 22 to work. Of course, there is the question of what breaks in the spago build if bumping up any dependency.

**Comment by FranklinChen** [comment]
For example, bumping up `glob` resulted in 

```
✘ [ERROR] No matching export in "node_modules/glob/dist/esm/index.js" for import "default"

    output/Docs.Search.IndexBuilder/foreign.js:1:7:
      1 │ import globMain from "glob";
        ╵        ~~~~~~~~
```

**Comment by f-f** [comment]
Thanks for looking into this! I think we'd be happy to upgrade the non-problematic dependencies, especially`better-sqlite3` if it allows us to work on node22 out of the box - anything goes as long as tests pass 😄

**Comment by f-f** [comment]
Fixed in #1230
